src/main/java/uk/ac/manchester/beehive/tornado/plugins/dynamicInspection/CodeGenerator.java

```diff
@@ -31,22 +30,23 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.HashMap;
-
+import java.util.Map;
 public class CodeGenerator {
 
-    public static void fileCreationHandler(Project project, ArrayList<PsiMethod> methods, String importCodeBlock) throws IOException {
+    public static void fileCreationHandler(Project project, ArrayList<PsiMethod> methods, ArrayList<PsiMethod> others, Map<String, Object> fields, String importCodeBlock) throws IOException {
         HashMap<String, PsiMethod> methodFile = new HashMap<>();
         File dir = FileUtilRt.createTempDirectory("files", null);
         for (PsiMethod method : methods) {
             String fileName = method.getName() + RandomStringUtils.randomAlphanumeric(5);
             File file = creatFile(project, method, importCodeBlock, fileName, dir);
+            File file = creatFile(project, method, others, importCodeBlock, fileName, dir, psiFile);
```

Comment:
@SarpHarbali can you have a look? 

This line gives the following error:
```bash
error: method creatFile in class CodeGenerator cannot be applied to given types;
            File file = creatFile(project, method, importCodeBlock, fileName, dir);
                        ^
  required: Project,PsiMethod,ArrayList<PsiMethod>,Map<String,Object>,String,String,File
  found:    Project,PsiMethod,String,String,File
  reason: actual and formal argument lists differ in length
```

Reply:
@stratika Fixed